### PR TITLE
memcache: lookup memcached servers port only on local node (SOC-10173)

### PR DESCRIPTION
**Pull request by dirkmueller**
With #2073 merged
we were looking up the port on the node with the lowest mac address,
which however in compilation phase might not yet have that attribute.

By passing in the local node and moving to execution phase we should
always have the default attributes for memcached port visible.

### CODE REVIEW

**Comment by dirkmueller** [comment]
@toabctl the underlying problem is that the get_memcached_servers() code is stupid in the sense that it just picks memcached_nodes.first and looks up an attribute there. however it is not guaranteed to have read after write consistency for nodes other than the $currentnode, so selecting first() might fundamentally fail, always, by design of how chef works.

**Comment by dirkmueller** [comment]
looks like this is actually a recent regression introduced by https://github.com/crowbar/crowbar-openstack/pull/2073

we might want to just revert that patch instead.

**Comment by stefannica** [comment]
Was jut looking at #2073 ...

File: chef/cookbooks/keystone/recipes/server.rb [hunk 1]
```diff
@@ -262,7 +258,8 @@
         node[:keystone][:api][:protocol],
         my_admin_host, node[:keystone][:api][:admin_port]
       ),
-      memcached_servers: memcached_servers,
+      memcached_servers: MemcachedHelper.get_memcached_servers(node,
```
Comment (stefannica) [inline chef/cookbooks/keystone/recipes/server.rb]: you missed one inline condition here

File: chef/cookbooks/nova/recipes/config.rb [hunk 2]
```diff
@@ -388,7 +383,8 @@
     vncproxy_cert_file: api_novnc_ssl_certfile,
     vncproxy_key_file: api_novnc_ssl_keyfile,
     serialproxy_public_host: public_api_host,
-    memcached_servers: memcached_servers,
+    memcached_servers:  MemcachedHelper.get_memcached_servers(node,
```
Comment (stefannica) [inline chef/cookbooks/nova/recipes/config.rb]: another one here

File: chef/cookbooks/keystone/recipes/server.rb (hunk 1 above)
Comment (dirkmueller) [reply chef/cookbooks/keystone/recipes/server.rb]: there is no inline condition here?

File: chef/cookbooks/nova/recipes/config.rb (hunk 2 above)
Comment (dirkmueller) [reply chef/cookbooks/nova/recipes/config.rb]: done

**Pull request merged**